extension/framework/goplugin/runner/reporter.go

```diff
@@ -0,0 +1,195 @@
+}
+
+func (reporter *Reporter) BeforeSuiteDidRun(setupSummary *types.SetupSummary) {
+}
+
+func (reporter *Reporter) SpecWillRun(specSummary *types.SpecSummary) {
+}
+
+func (reporter *Reporter) SpecDidComplete(specSummary *types.SpecSummary) {
+	reporter.specs = append(reporter.specs, deepcopy.Copy(*specSummary).(types.SpecSummary))
+}
+
+func (reporter *Reporter) AfterSuiteDidRun(setupSummary *types.SetupSummary) {
+}
+
+func (reporter *Reporter) SpecSuiteDidEnd(summary *types.SuiteSummary) {
+	for i, _ := range reporter.suites {
+		if reporter.suites[i].SuiteDescription == summary.SuiteDescription {
+			reporter.suites[i] = deepcopy.Copy(*summary).(types.SuiteSummary)
+			break
+		}
+	}
+}
+
+func (reporter *Reporter) Prepare(description string) {
+	reporter.suites = append(reporter.suites, types.SuiteSummary{
+		SuiteDescription: description,
+		SuiteSucceeded:   true,
+		SuiteID:          "undefined",
+		NumberOfSpecsBeforeParallelization: 0,
+		NumberOfTotalSpecs:                 0,
+		NumberOfSpecsThatWillBeRun:         0,
+		NumberOfPendingSpecs:               0,
+		NumberOfSkippedSpecs:               0,
+		NumberOfPassedSpecs:                0,
+		NumberOfFailedSpecs:                0,
+		RunTime:                            time.Duration(0),
+	})
+}
+
+func (reporter *Reporter) Report() {
+	fmt.Println("--------------------------------------------------------------------------------")
+
+	fmt.Println("Failures:")
+	fmt.Println()
+
+	steno := stenographer.New(true)
+
+	for _, spec := range reporter.specs {
+		steno.AnnounceCapturedOutput(spec.CapturedOutput)
+
+		switch spec.State {
+		case types.SpecStatePassed:
+			if spec.IsMeasurement {
+				steno.AnnounceSuccesfulMeasurement(&spec, configSuccinct)
+			} else if spec.RunTime.Seconds() >= configSlowSpecThreshold {
+				steno.AnnounceSuccesfulSlowSpec(&spec, configSuccinct)
+			} else {
+				steno.AnnounceSuccesfulSpec(&spec)
+			}
+
+		case types.SpecStatePending:
+			steno.AnnouncePendingSpec(&spec, configNoisyPendings && !configSuccinct)
+		case types.SpecStateSkipped:
+			steno.AnnounceSkippedSpec(&spec, configSuccinct, configFullTrace)
+		case types.SpecStateTimedOut:
+			steno.AnnounceSpecTimedOut(&spec, configSuccinct, configFullTrace)
+		case types.SpecStatePanicked:
+			steno.AnnounceSpecPanicked(&spec, configSuccinct, configFullTrace)
+		case types.SpecStateFailed:
+			steno.AnnounceSpecFailed(&spec, configSuccinct, configFullTrace)
+		}
+	}
+
+	fmt.Println()
+	fmt.Println("Report:")
+	fmt.Println()
+
+	// total
+	totalNumberOfTotalSpecs := 0
+	totalNumberOfPassedSpecs := 0
+	totalNumberOfFailedSpecs := 0
+	totalNumberOfSkippedSpecs := 0
+	totalNumberOfPendingSpecs := 0
+	totalRunTime := time.Duration(0) * time.Nanosecond
+
+	// suites
+	for index, suite := range reporter.suites {
+		fmt.Printf("[%4d] ", index+1)
+		if suite.NumberOfFailedSpecs == 0 {
+			fmt.Printf(greenColor+"%-80s"+defaultStyle, suite.SuiteDescription)
```

Comment:
width of fields is too generous: even though I use widescreen terminal, I'm getting wrapped lines. 60-70 chars for test binary + 5-6 chars for number of test cases failures/successes should be fine.